policy/gql_mutations.py

```diff
@@ -50,6 +50,20 @@ def do_mutate(cls, perms, user, **data):
         from core.utils import TimeUtils
         data['validity_from'] = TimeUtils.now()
         policy = PolicyService(user).update_or_create(data, user)
+        logger.info(f"After policy create_or_update: {policy.uuid}")
+        if data["stage"] == Policy.STAGE_RENEWED:
+            logger.info("Deleting the optional PolicyRenewals after renewing")
+            previous_policy = (Policy.objects.filter(validity_to__isnull=True,
+                                                     family_id=data["family_id"],
+                                                     product_id=data["product_id"],
+                                                     status__in=[Policy.STATUS_EXPIRED, Policy.STATUS_ACTIVE])
+                                             .order_by("-id")
+                                             .first())
+            if not previous_policy:
+                logger.error("Can't find the policy that was renewed - not deleting the PolicyRenewals")
+            policy_renewals = PolicyRenewal.objects.filter(policy=previous_policy, validity_to__isnull=True)
```

Comment:
Should it return something before if there is no previous policy ?

Reply:
It depends. It's happening after the policy renewal is created so they just igrone if there is no previous policy.Should that cause the whore renewal to be rejected?

Reply:
Let's keep it as such for now and we'll wait for implementation request to change this.